qemu/tests/seabios_bin.py

```diff
@@ -0,0 +1,58 @@
+import logging
+
+from virttest import error_context
+from virttest import env_process
+from virttest import utils_misc
+
+from avocado.utils import process
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    KVM Seabios bin file test:
+    1) Get available bin files
+    2) Get supported machine types
+    3) Check bin file in use
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    bin_dict = {'rhel6': 'bios.bin', 'rhel7': 'bios-256k.bin'}
+
+    error_context.context("Get available bin files", logging.info)
+    output = process.system_output('ls /usr/share/seabios', shell=True)
+    for value in bin_dict.values():
+        if value not in output:
+            test.fail("%s is not available" % value)
+
+    error_context.context("Get supported machine types", logging.info)
+    qemu_binary = utils_misc.get_qemu_binary(params)
+    machine_type_cmd = qemu_binary + " -machine help | awk '{ print $1 }'"
+    output = process.system_output(machine_type_cmd, shell=True)
+    machine_types = output.splitlines()[1:-1]
+    machine_types.remove('pc')
+    machine_types.remove('q35')
+    logging.info(machine_types)
+
+    for machine in machine_types:
+        error_context.context("Check bin file with machine type: %s" % machine,
+                              logging.info)
+        for key in bin_dict:
+            if key in machine:
+                bin_file = bin_dict[key]
+                break
+        else:
+            raise NotImplementedError()
```

Comment:
You mean `test.cancel()` here?

Reply:
No, this part will also be updated when the supported machine types of rhel8 are concluded.